test/test_testing.py

```diff
@@ -1274,9 +1277,9 @@ def test_hooks_disabled_noop(self):
         harness.update_config({'eighth': '8'})
         assert harness.charm.get_changes(reset=True) == []
 
-    def test_metadata_from_directory(self):
+    def test_metadata_from_directory(self, request: pytest.FixtureRequest):
         tmp = pathlib.Path(tempfile.mkdtemp())
-        self.addCleanup(shutil.rmtree, str(tmp))
+        request.addfinalizer(lambda: shutil.rmtree(tmp))
```

Comment:
Originally I was in favour of the `tmpdir` fixture, but since there are a bunch of test cases like this (6 in this file of this kind, many more in other files), using the `tmpdir` fixture means adding another argument in the arg list for many test functions to do something the first fixture can already do, so the team want to use as fewer fixtures as possible.